packages/autorest.go/src/generator/helpers.ts

```diff
@@ -575,6 +575,12 @@ export async function getParentImport(session: Session<CodeModel>): Promise<stri
   const modName = await session.getValue('module', 'none');
   const containingMod = await session.getValue('containing-module', 'none');
   if (modName !== 'none') {
+    const majorVersion = (await session.getValue('module-version', '1.0.0')).split('.')[0];
+    if (Number(majorVersion) > 1) {
+      if (modName.split('/')[-1] !== 'v' + majorVersion) {
+        return modName + '/v' + majorVersion;
```

Comment:
This is going to double up the major version number if one already exists in the module name ([example](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/compute/armcompute/autorest.md)).

If we want the contract to be that `module` must _not_ contain a major version suffix then we need to verify that when reading its value and throw and error if it does.

Reply:
Line 580 has handled such situation: if the module name has vX suffix, then use it directly.

Reply:
Hi @jhendrixMSFT, modified according to above discussion, please review.